Run unit tests on three operating systems

hugovk opened the pull request:
After removing sed in https://github.com/python/docsbuild-scripts/pull/272, we should now support Windows in addition to Linux and macOS?

Let's test them to avoid regressions.

AA-Turner commented:
This doesn't exercise the `sed` code. If we want to test on multiple OSes then perhaps prefer adding it to the integration test?

A

hugovk commented:
Yeah, the integration test was failing when I opened the PR and fixed a couple of minutes later :) Updated, also added 3.14.

There is also value running the unit tests on the other operating systems, we're able to break code there too, whether it's exercises the sed code or not.

ezio-melotti commented on .github/workflows/test.yml:
```diff
@@ -13,14 +13,24 @@ env:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+        - "3.13"
+        - "3.14"
+        os: [windows-latest, macos-latest, ubuntu-latest]
```
```suggestion
        python-version: ["3.13", "3.14"]
        os: [windows-latest, macos-latest, ubuntu-latest]
```
Shorter and consistent with the `os`?

ezio-melotti commented on .github/workflows/test.yml:
```diff
@@ -41,13 +51,14 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
         python-version:
         - "3.13"
         - "3.14"
+        os: [windows-latest, macos-latest, ubuntu-latest]
```
```suggestion
        python-version: ["3.13", "3.14"]
        os: [windows-latest, macos-latest, ubuntu-latest]
```
Is there a way to remove this duplication?

rffontenelle replied:
Dynamic matrix values could be used, which consists in a first job setting up matrices' values, making them as output values, and then the actual jobs use those outputs as matrix values. See e.g. [this stackoverflow answered question](https://stackoverflow.com/questions/65056670/is-it-possible-to-have-a-dynamic-strategy-matrix-in-a-workflow-in-github-actions).

hugovk replied:
I don't think the extra complexity is worth it.

hugovk commented:
Integration tests were silently passing even when really failing -- that was fixed in https://github.com/python/docsbuild-scripts/pull/286 and so they do properly fail when failing now.

And the integration tests were really failing here on Windows ("FileNotFoundError: [WinError 2] The system cannot find the file specified") and macOS ("rsync: --delete-delay: unknown option") and 3.14 ("ERROR: Failed building wheel for pillow"), so for now let's keep this PR just for unit tests after all.
